staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -702,19 +729,6 @@ func (r *Request) request(fn func(*http.Request, *http.Response)) error {
 		metrics.RequestLatency.Observe(r.verb, r.finalURLTemplate(), time.Since(start))
 	}()
 
-	if r.err != nil {
-		klog.V(4).Infof("Error in request: %v", r.err)
-		return r.err
-	}
-
-	// TODO: added to catch programmer errors (invoking operations with an object with an empty namespace)
```

Comment:
You can call the same function again here.  I don't want any future `Do` or `DoRaw` to have an accident and this is cheap.

Reply:
I added a call to `requestPreflightCheck()` here, coming up on the next commit.